### PR TITLE
Feature/rfhub2 8 import standalone libs properly

### PR DESCRIPTION
**Pull request by MaciejWiczk**
Feature changing libraries logic: 
* load Libraries with __init__ as single library
* load single file library without dots in name (ie. ClassFour instead
of ClassFour.ClassFour)
* single file libraries with empt __init__ file are now imported as
single libraries
* classes different than filename are now ommited
* added fixtures for tests

How RfHub2 used to be:
rfhub2

How it is now:
rfhub2-fix

### CODE REVIEW

File: rfhub2/kwdb.py [hunk 1]
```diff
@@ -218,7 +196,10 @@ def add_folder(self, dirname, watch=True):
                 if os.path.isdir(path):
                     if not basename.startswith("."):
                         if os.access(path, os.R_OK):
-                            self.add_folder(path, watch=False)
+                            if self._looks_like_library_with_init(path):
```
Comment (pbylicki) [inline rfhub2/kwdb.py:200]: Did you consider scenario where calling method LibraryDocumentation inside this method raises DataError and else branch is omitted? Is it realistic?

File: tests/fixtures/LibWithInit/ClassOne.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+class ClassOne(object):
+    """
+    Overview that should not be imported for ClassOne.
+    """
+    def __init__(self):
+        self.a = None
+
+    def class_1_method_1(self):
+        """Docsting for class_1_method_1"""
```
Comment (pbylicki) [inline tests/fixtures/LibWithInit/ClassOne.py]: Spelling error - should be Docstring. Analogically in the rest of added test classes

**Comment by pbylicki** [comment]
How about adding some tests checking that all required keywords were loaded from the test libraries with various setup (those that you listed above)?

File: rfhub2/kwdb.py (hunk 1 above)
Comment (MaciejWiczk) [reply rfhub2/kwdb.py:200]: I will check that. If needed I will add some tests and resources for it.

File: tests/fixtures/LibWithInit/ClassOne.py (hunk 2 above)
Comment (MaciejWiczk) [reply tests/fixtures/LibWithInit/ClassOne.py]: Noted and fixed

File: rfhub2/kwdb.py (hunk 1 above)
Comment (MaciejWiczk) [reply rfhub2/kwdb.py:200]: How about wrapping it in try, except:
~~~~{.python}
        try:
            return len(LibraryDocumentation(path).keywords) > 0 \
                if os.path.isfile(os.path.join(path, '__init__.py')) else False
        except DataError:
            return False
~~~~
I don't know how we could end up raising DataError, but better be safe than sorry.

**Comment by MaciejWiczk** [comment]
> How about adding some tests checking that all required keywords were loaded from the test libraries with various setup (those that you listed above)?

I've added three tests covering doc page. I didn't use API, but it will be good to create some neat tests for that, when changing to ORM

File: rfhub2/kwdb.py [hunk 3]
```diff
@@ -512,8 +513,14 @@ def _looks_like_library_file(self, name):
...
     def _looks_like_library_with_init(self, path):
-        return len(LibraryDocumentation(path).keywords) > 0 \
-            if os.path.isfile(os.path.join(path, '__init__.py')) else False
+        try:
+            return len(LibraryDocumentation(path).keywords) > 0 \
+                if os.path.isfile(os.path.join(path, '__init__.py')) else False
+        except DataError:
+            return False
+
+        # return len(LibraryDocumentation(path).keywords) > 0 \
```
Comment (pbylicki) [inline rfhub2/kwdb.py]: Commented part can be removed

File: rfhub2/kwdb.py [hunk 4]
```diff
@@ -512,8 +513,14 @@ def _looks_like_library_file(self, name):
         return name.endswith(".py")
 
     def _looks_like_library_with_init(self, path):
-        return len(LibraryDocumentation(path).keywords) > 0 \
-            if os.path.isfile(os.path.join(path, '__init__.py')) else False
+        try:
+            return len(LibraryDocumentation(path).keywords) > 0 \
```
Comment (pbylicki) [inline rfhub2/kwdb.py]: can be written as `return os.path.isfile(os.path.join(path, '__init__.py')) and len(LibraryDocumentation(path).keywords) > 0` to skip if and explicit False

File: rfhub2/kwdb.py (hunk 3 above)
Comment (MaciejWiczk) [reply rfhub2/kwdb.py]: Done

File: rfhub2/kwdb.py (hunk 4 above)
Comment (MaciejWiczk) [reply rfhub2/kwdb.py]: Done

**Review by pbylicki — APPROVED** [review]

**Pull request merged**